target_snowflake/connector.py

```
@@ -349,10 +349,11 @@ def _get_merge_from_stage_statement(
     def _get_copy_statement(self, full_table_name, schema, sync_id, file_format):
         """Get Snowflake COPY statement."""
         formatter = SnowflakeIdentifierPreparer(SnowflakeDialect())
+        formatted_properties = ", ".join([formatter.format_collation(col) for col in schema["properties"].keys()])
         column_selections = self._get_column_selections(schema, formatter)
         return (
             text(
-                f"copy into {full_table_name} from "
+                f"copy into {full_table_name} ({formatted_properties}) from "
```

Should this be `column_selections` ? You want it to always match the snowflake table in `full_table_name` right? 

My assumption is there's column checks earlier to make seure `full_table_name` has the columns setup alredy before this runs

---

What we did here https://github.com/MeltanoLabs/target-postgres/blob/main/target_postgres/sinks.py#L194 was just used the columns directly from the source table (the stage in this case). I wonder if that'd be even better here

---

@visch `column_selection` is a formatted list used to cast to types, etc. https://github.com/MeltanoLabs/target-snowflake/blob/f560958da71d1f643428e64a70e3ae38a79b80c7/target_snowflake/connector.py#L309. This is just a list of column names.

Selecting the columns from the table before building the query is more definitive but if they didnt match then theres a bug elsewhere in the table creation or altering that happens before this step. Although couldnt that lead to us silently skipping columns if theres a bug that is failing to all the columns? Am I understanding correctly?

---

My concern is that `column_selections` generates columns in a seperate function to `formatted_properties` right now they loop over the same schema in the same way which means order is preserved (which is the key here) but how do you keep that going in the long term so that folks don't mess that up. 

Maybe not a huge deal, but it's something that nags at me as I've seen this kind of thing blow up later as people forget the context of why things were done a certain way

---

@visch oh yeah it adds some duplicate code that could be pulled out and reused. I'll see if I can refactor quickly, otherwise I'll open a separate issue so we can squash the bug and clean up after.

---

I like it better! Long term sqlalchmy tables / columns would probably be the slickest but that's too much lift for sure!